base_search_improved/models/base_model.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# © 2016 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, api
+
+
+class BaseModelPatch(models.Model):
+    _name = "base.model.patch"
+    _auto = False
+
+    def init(self, cr):
+
+        base_model = BaseModelPatch.__bases__[0]
+        upgrade = lambda self, value, args, offset=0, limit=None, \
+            order=None, count=False: value if count else self.browse(value)
+        downgrade = lambda self, value, args, offset=0, limit=None, \
+            order=None, count=False: value if count else value.ids
+
+        @api.returns('self', upgrade=upgrade, downgrade=downgrade)
+        def new_search(self, cr, user, args, offset=0, limit=None,
+                       order=None, context=None, count=False):
+            for i, cond in enumerate(args):
+                # fuzzy search: replace spaces with wildcards
+                if len(cond) == 3 and cond[1] == 'ilike':
+                    args[i] = cond[0], cond[1], cond[2].replace(' ', '%')
+            return new_search.origin(self, cr, user, args, offset, limit,
```

Comment:
What about the following special case: [Search by invoice.number](https://github.com/odoo/odoo/blob/5f9ca1b89becb6867d0755cf7cb3d7f9c533e7af/addons/account/models/account_invoice.py#L874-L877)

- Data:
 - invoice1: 123 987
 - invoice2: 123 789 987

- Original behaviour:
 - Search: "123 987" returns invoice1

- New behaviour:
 - Search: "123 987" returns invoice1 and invoice2